sos/cleaner/obfuscation_archive.py

```diff
@@ -119,29 +135,40 @@ def compress(self, cmd):
         res = sos_get_command_output(exec_cmd, timeout=0, stderr=True)
         if res['status'] == 0:
             self.final_archive_path = self.tarpath + '.' + exec_cmd[0:2]
+            self.log_debug("Compressed to %s" % self.final_archive_path)
+            try:
+                self.remove_extracted_path()
+            except Exception as err:
+                self.log_debug("Failed to remove extraction directory: %s"
+                               % err)
+                self.report_msg('Failed to remove temporary extraction '
+                                'directory')
         else:
             err = res['output'].split(':')[-1]
             self.log_debug("Exception while compressing archive: %s" % err)
             raise Exception(err)
 
-    def generate_checksum(self, hash_name):
-        """Calculate a new checksum for the obfuscated archive, as the previous
-        checksum will no longer be valid
+    def remove_extracted_path(self):
+        """After the tarball has been re-compressed, remove the extracted path
+        so that we don't take up that duplicate space any longer during
+        execution
         """
-        try:
-            hash_size = 1024**2  # Hash 1MiB of content at a time.
-            archive_fp = open(self.final_archive_path, 'rb')
-            digest = hashlib.new(hash_name)
-            while True:
-                hashdata = archive_fp.read(hash_size)
-                if not hashdata:
-                    break
-                digest.update(hashdata)
-            archive_fp.close()
-            return digest.hexdigest()
-        except Exception as err:
-            self.log_debug("Could not generate new checksum: %s" % err)
-        return None
+        def force_delete_file(action, name, exc):
+            os.chmod(name, stat.S_IWUSR)
+            if os.path.isfile(name):
+                os.remove(name)
+            else:
+                shutil.rmtree(name)
+        if os.getuid() != 0:
+            # this is rather silly, but due to the fact we preserve permissions
+            # in reports, a non-root user would be unable to remove a lot of
+            # paths under proc/ and sys/. Since we're at the point where the
+            # archive has been re-compressed and we're removing data that will
+            # no longer be referenced anywhere, just open the perms to allow
+            # us to delete it all
+            sos_get_command_output("chmod -R 777 %s" % self.extracted_path)
```

Comment:
This is the chmod 777 I'm most worried about.  chmod can end up taking a while.. and then it's all open unless I'm missing somethng.  I also don't full grep why a full 777 helps with removal.

Reply:
OH, I thought you were referring to the extraction permission changes.

That chmod was added before I resolved to add the execute bit for directory perms on extraction. It may be possible to drop it at this point, let me run some tests.

Reply:
Ack, we should be able to drop this now.